plugin/pkg/scheduler/scheduler.go

```diff
@@ -252,6 +252,7 @@ func (sched *Scheduler) assumeAndBindVolumes(assumed *v1.Pod, host string) error
 	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeScheduling) {
 		allBound, bindingRequired, err := sched.config.VolumeBinder.Binder.AssumePodVolumes(assumed, host)
 		if err != nil {
+			assumed = assumed.DeepCopy()
```

Comment:
@bsalamat instead of doing the DeepCopy() here, I'm thinking it might be better to do it outside in scheduleOne() here: 
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/scheduler.go#L325

WDYT?

Reply:
Makes sense.

Reply:
The sequence of events to trigger this issue was:
1. Pod fails predicates, pod condition was updated with the predicate error.
2. Pod passes predicates
3. Pod fails bind, pod condition was updated with the bind error.  It seems like the Conditions in `assumedPod` is the same ptr as in `pod` because `assumedPod` is not a DeepCopy of `pod`.

Reply:
Updated, let me know when I can squash

Reply:
Thanks! Looks good. Please go ahead and squash.

Reply:
You actually fixed a bug that existed in the scheduler code for a while. The scenario that you described could happen without your changes (and in fact, we had seen some api cache panics which I think was caused by this issue).